smartlearner/optimizers/adam.py

```diff
@@ -0,0 +1,43 @@
+from collections import OrderedDict
+
+import theano
+import theano.tensor as T
+import numpy as np
+
+from ..interfaces import Optimizer
+from ..utils import sharedX
+
+class Adam(Optimizer):
```

Comment:
I agree, it probably doesn't matter. And subclassing directly `Optimizer` has the advantage of being clearer (i.e. that we use the opposite of the gradient). Like I said `AdaGrad`'s code dates from when it was implemented as an `DirectionModifier`/`UpdateRule`. It is fine with me if we adapt it to better fit the current framework.